shared-libs/transitions/src/transitions/index.js

```diff
@@ -0,0 +1,331 @@
+const _ = require('underscore'),
+      async = require('async'),
+      db = require('../db'),
+      lineage = require('@medic/lineage')(Promise, db.medic),
+      utils = require('../lib/utils'),
+      logger = require('../lib/logger'),
+      config = require('../config'),
+      infodoc = require('../lib/infodoc'),
+      uuid = require('uuid');
+
+
+/*
+ * Add new transitions here to make them available for configuration and execution.
+ * Transitions are executed in the order they appear in this array.
+ *
+ * (For security reasons, we want to avoid doing a `require` based on random input,
+ *  hence we maintain this index of transitions.)
+ *
+ * Transition on this list:
+ *  - Are disabled by default, and can be enabled in configuration
+ *  - should be documented as to what they do and how they are configured in
+ *    medic-docs
+ */
+const AVAILABLE_TRANSITIONS = [
+  'update_clinics',
+  'registration',
+  'accept_patient_reports',
+  'generate_patient_id_on_people',
+  'default_responses',
+  'update_sent_by',
+  'update_sent_forms',
+  'death_reporting',
+  'conditional_alerts',
+  'multi_report_alerts',
+  'update_notifications',
+  'update_scheduled_reports',
+  'resolve_pending',
+  'muting'
+];
+
+const transitions = [];
+
+/*
+ * Load transitions using `require` based on what is in AVAILABLE_TRANSITIONS
+ * constant and what is enabled in the `transitions` property in the settings
+ * data.  Log warnings on failure.
+ */
+const loadTransitions = (synchronous = false) => {
+  const self = module.exports;
+  const transitionsConfig = config.get('transitions') || [];
+  transitions.splice(0, transitions.length);
+  let loadError = false;
+
+  // Load all configured transitions
+  AVAILABLE_TRANSITIONS.forEach(transition => {
+    const conf = transitionsConfig[transition];
+
+    const disabled = conf && conf.disable;
+    const allowSync = conf && !conf.async;
```

Comment:
I chose to make the config "disable" some transitions from running synchronously instead of "enabling" to run synchronously (after back and fourth-ing  a few times). Reasoning was to enable projects to start using this feature without changes to configurations.
AFAICT the only transitions which **should** run async are `muting` and `update_notifications`, which may end up needing to updated lots of documents.

I would appreciate some feedback on this.